setup.py

```diff
@@ -190,390 +98,217 @@ def check_dependency_versions(min_versions):
-        GIT_REVISION = git_version()
-    elif os.path.exists(filename):
-        # must be a source distribution, use existing version file
-        try:
-            from statsmodels.version import git_revision as GIT_REVISION
-        except ImportError:
-            dowrite = False
-            GIT_REVISION = "Unknown"
-    else:
-        GIT_REVISION = "Unknown"
+ADDITIONAL_PACKAGE_DATA = {
+    'statsmodels.datasets.tests': ['*.zip'],
+    'statsmodels.iolib.tests.results': ['*.dta'],
+    'statsmodels.stats.tests.results': ['*.json'],
+    'statsmodels.tsa.vector_ar.tests.results': ['*.npz'],
+    'statsmodels.stats.tests': ['*.txt'],
+    'statsmodels.stats.libqsturng': ['*.r', '*.txt', '*.dat'],
+    'statsmodels.stats.libqsturng.tests': ['*.csv', '*.dat'],
+    'statsmodels.tsa.vector_ar.data': ['*.dat'],
+    'statsmodels.sandbox.regression.tests': ['*.dta', '*.csv']
+}
+
+##############################################################################
+# Extension Building
+##############################################################################
+COMPILER_DIRECTIVES = {}
+DEFINE_MACROS = []
+
+
+exts = dict(
+    _exponential_smoothers={'source': 'statsmodels/tsa/_exponential_smoothers.pyx'},  # noqa: E501
+    _hamilton_filter={'source': 'statsmodels/tsa/regime_switching/_hamilton_filter.pyx.in'},  # noqa: E501
+    _kim_smoother={'source': 'statsmodels/tsa/regime_switching/_kim_smoother.pyx.in'},  # noqa: E501
+    linbin={'source': 'statsmodels/nonparametric/linbin.pyx'},
+    _smoothers_lowess={'source': 'statsmodels/nonparametric/_smoothers_lowess.pyx'},  # noqa: E501
+    kalman_loglike={'source': 'statsmodels/tsa/kalmanf/kalman_loglike.pyx',
+                    'include_dirs': ['statsmodels/src'],
+                    'depends': ['statsmodels/src/capsule.h']})
+
+statespace_exts = [
+    'statsmodels/tsa/statespace/_initialization.pyx.in',
+    'statsmodels/tsa/statespace/_representation.pyx.in',
+    'statsmodels/tsa/statespace/_kalman_filter.pyx.in',
+    'statsmodels/tsa/statespace/_filters/_conventional.pyx.in',
+    'statsmodels/tsa/statespace/_filters/_inversions.pyx.in',
+    'statsmodels/tsa/statespace/_filters/_univariate.pyx.in',
+    'statsmodels/tsa/statespace/_filters/_univariate_diffuse.pyx.in',
+    'statsmodels/tsa/statespace/_kalman_smoother.pyx.in',
+    'statsmodels/tsa/statespace/_smoothers/_alternative.pyx.in',
+    'statsmodels/tsa/statespace/_smoothers/_classical.pyx.in',
+    'statsmodels/tsa/statespace/_smoothers/_conventional.pyx.in',
+    'statsmodels/tsa/statespace/_smoothers/_univariate.pyx.in',
+    'statsmodels/tsa/statespace/_smoothers/_univariate_diffuse.pyx.in',
+    'statsmodels/tsa/statespace/_simulation_smoother.pyx.in',
+    'statsmodels/tsa/statespace/_tools.pyx.in',
+]
+
+
+class CleanCommand(clean):
+    def run(self):
+        msg = """
 
-    if not ISRELEASED:
-        FULLVERSION += '.dev0+' + GIT_REVISION[:7]
+python setup.py clean is not supported.
```

Comment:
Why not?

Reply:
It needs a custom clean to be useful, but this is fragile and adds a lot of LOC.  Git clean -xfd is a simple method to clean everything not tracked.

Reply:
It is also what NumPy (and possibly pandas) does.

Reply:
pandas supports it; not a big enough deal to risk de-railing this PR